clang/include/clang/Driver/Options.td

```diff
@@ -3120,7 +3120,16 @@ def emit_cir_only : Flag<["-"], "emit-cir-only">,
 def emit_cir_flat : Flag<["-"], "emit-cir-flat">, Visibility<[ClangOption, CC1Option]>,
   Group<Action_Group>, HelpText<"Similar to -emit-cir but also lowers structured CFG into basic blocks.">;
 def emit_mlir : Flag<["-"], "emit-mlir">, Visibility<[CC1Option]>, Group<Action_Group>,
-  HelpText<"Build ASTs and then lower through ClangIR to MLIR, emit the .milr file">;
+  HelpText<"Build ASTs and then lower through ClangIR to MLIR (standard dialects "
+  "when `-fno-clangir-direct-lowering` is used or the LLVM dialect when "
+  "`-fclangir-direct-lowering` is used), emit the .mlir file.">;
+def emit_mlir_EQ : Joined<["-"], "emit-mlir=">, Visibility<[CC1Option]>, Group<Action_Group>,
+  HelpText<"Build ASTs and then lower through ClangIR to the selected MLIR dialect, emit the .mlir file. "
+  "Allowed values are `std` for MLIR standard dialects and `llvm` for the LLVM dialect.">,
+  Values<"std,llvm">,
```

Comment:
Can "cir" and "cir-flat" be added here?

Reply:
I asked to remove it but now in light of upstreaming discussion https://github.com/llvm/llvm-project/pull/127835#discussion_r1964324860, we should probably converge to make upstreaming approach easier. @Jezurko sorry for the noise. We should add `cir` and `cir-flat` to match was likely coming for Flang as part of that discussion. We should also rename `std` to `core`.

Andy, should we hold on this until you land to make your life easier or is it enough if @Jezurko applies the changes above?

Reply:
I think it makes sense to merge this first, then I can update my upstream patch to align with this and the wishes of the reviewers there.

Reply:
I have posted an update in https://github.com/llvm/llvm-project/pull/127835 that uses the `-emit-mlir=[cir|core]` option. I was referencing this PR when I implemented it, so it shouldn't be too far off. One difference is that I made the MLIR dialect a separate named enum.

Reply:
Added another small discussion point over there